### PR TITLE
Build with ESBuild

### PR DESCRIPTION
**Pull request by nawatts**
Replace Babel with [ESBuild](https://esbuild.github.io/).

First builds take longer because I ran `rm -r node_modules/.cache`.

## Before

```
$ for i in $(seq 1 5); do { time yarn run build >/dev/null 2>&1 } 2>&1; done
yarn run build > /dev/null 2>&1  128.85s user 5.98s system 200% cpu 1:07.26 total
yarn run build > /dev/null 2>&1  66.44s user 3.43s system 337% cpu 20.715 total
yarn run build > /dev/null 2>&1  66.04s user 3.42s system 340% cpu 20.427 total
yarn run build > /dev/null 2>&1  61.20s user 3.42s system 304% cpu 21.192 total
yarn run build > /dev/null 2>&1  64.05s user 3.87s system 315% cpu 21.541 total
```

## After

```
$ for i in $(seq 1 5); do { time yarn run build >/dev/null 2>&1 } 2>&1; done
yarn run build > /dev/null 2>&1  75.39s user 3.32s system 161% cpu 48.595 total
yarn run build > /dev/null 2>&1  29.35s user 2.24s system 192% cpu 16.423 total
yarn run build > /dev/null 2>&1  29.40s user 2.27s system 195% cpu 16.238 total
yarn run build > /dev/null 2>&1  29.59s user 2.37s system 195% cpu 16.329 total
yarn run build > /dev/null 2>&1  29.63s user 2.33s system 194% cpu 16.418 total
```

### CODE REVIEW

File: src/components/WDLViewer.js [hunk 1]
```diff
@@ -1,5 +1,11 @@
+import 'prismjs'
+import 'prismjs/components/prism-bash'
+import 'prismjs/components/prism-python'
+import 'prismjs/plugins/line-numbers/prism-line-numbers'
+import 'prismjs/plugins/line-numbers/prism-line-numbers.css'
+import 'prismjs/themes/prism.css'
+
 import _ from 'lodash/fp'
-import Prism from 'prismjs'
```
Comment (nawatts) [inline src/components/WDLViewer.js:2]: This was dependent on a Babel plugin.

**Comment by nawatts** [comment]
Looks like yarn run locally cached `esbuild-darwin-arm64` whereas CircleCI wants `esbuild-linux-64`.
https://app.circleci.com/pipelines/github/DataBiosphere/terra-ui/11841/workflows/7d5dc398-8163-4dba-92e9-6f010d106f4f/jobs/50341

**Comment by kyuksel** [comment]
FWIW quick hot reload times using MacBook Pro (16-inch, 2019) running Monterey (v12.5.1):

After removing the string `flexible` from the line

```javascript
getEntitiesTsv: entityType => fetchOrchestration(`api/workspaces/${namespace}/${name}/entities/${entityType}/tsv?model=flexible`, _.mergeAll([authOpts(), { signal }])).then(r => r.text())
```

in `ajax.js`

JS: `15606ms`
TS: `18240ms`

which looks promising 😊

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 FROM node:16
 
 COPY . /terra-ui/
-RUN cd /terra-ui && yarn build
+RUN cd /terra-ui && yarn install --immutable-cache && yarn build
```
Comment (nawatts) [inline Dockerfile:4]: ESBuild is "unplugged", so requires `yarn install`.

**Comment by nawatts** [comment]
Discussed at UIWG meeting yesterday. Not enough interest to pursue this right now.

**Pull request closed without merging**